### PR TITLE
Bug Fix : Convert labels for multi-label Classification into LongTensor before one_hot encoding

### PR DESCRIPTION
**Pull request by rajeshgangireddy**
### Summary

For multilabel classification, we use torch' one_hot to get one hot encoding of the labels. This method requires labels to be for dtype LongTensor. When we have empty labels, then the tensor is of type float32. This gives an error.  This PR manually changes the labels to be LongTensor if not already before calling the one_hot function. 

[one_hot](https://pytorch.org/docs/stable/generated/torch.nn.functional.one_hot.html#torch-nn-functional-one-hot)

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have ran e2e tests and there is no issues.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/open-edge-platform/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/open-edge-platform/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/open-edge-platform/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2025 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by sovrasov — DISMISSED** [review]

**Comment by sovrasov** [comment]
Does the problem occur only on samples with no labels?

**Comment by rajeshgangireddy** [comment]
@sovrasov 
> Does the problem occur only on samples with no labels?


Yes.

**Comment by eugene123tw** [comment]
Thanks Rajesh fixing this issue :100:  Would it be possible to change labels.to(dtype=torch.long, copy=False) to labels.long() instead? I think it's mainly for brevity and since performance or memory usage may not be a concern in this case as it's a small tensor.

**Comment by rajeshgangireddy** [comment]
> Thanks Rajesh fixing this issue 💯 Would it be possible to change labels.to(dtype=torch.long, copy=False) to labels.long() instead? I think it's mainly for brevity and since performance or memory usage may not be a concern in this case as it's a small tensor.

Done.

File: src/otx/core/data/dataset/classification.py [hunk 1]
```diff
@@ -111,6 +111,9 @@ def _get_item_impl(self, index: int) -> TorchDataItem | None:
 
     def _convert_to_onehot(self, labels: torch.tensor, ignored_labels: list[int]) -> torch.tensor:
         """Convert label to one-hot vector format."""
+        # Torch's one_hot() expects the input to be of type long
+        # However, when labels are empty, they are of type float32
+        labels = labels.long()
         onehot = functional.one_hot(labels, self.num_classes).sum(0).clamp_max_(1)
```
Comment (Daankrol) [inline src/otx/core/data/dataset/classification.py]: not sure if the comments are necessary. 
```suggestion
        onehot = functional.one_hot(labels.long(), self.num_classes).sum(0).clamp_max_(1)
```

**Review by Daankrol — APPROVED** [review]

**Review by sovrasov — APPROVED** [review]
